cargo-dist/src/lib.rs

```diff
@@ -395,13 +396,133 @@ fn write_checksum(checksum: &str, src_path: &Utf8Path, dest_path: &Utf8Path) ->
+                    line: line.to_owned(),
+                })
+            }
+        };
+
+        parsed.insert(key, value);
+    }
+
+    Ok(parsed)
+}
+
+// Takes a BTreeMap of key/value environment variables produced by
+// `brew bundle exec` and decides which ones we want to keep for our own builds.
+// Returns a Vec containing (KEY, value) tuples.
+fn select_brew_env(environment: &BTreeMap<&str, &str>) -> Vec<(String, String)> {
+    let mut desired_env = vec![];
+
+    // Several of Homebrew's environment variables are safe for us to use
+    // unconditionally, so pick those in their entirety.
+    if let Some(value) = environment.get("PKG_CONFIG_PATH") {
+        desired_env.push(("PKG_CONFIG_PATH".to_owned(), value.to_string()))
+    }
+    if let Some(value) = environment.get("PKG_CONFIG_LIBDIR") {
+        desired_env.push(("PKG_CONFIG_LIBDIR".to_owned(), value.to_string()))
+    }
+    if let Some(value) = environment.get("CMAKE_INCLUDE_PATH") {
+        desired_env.push(("CMAKE_INCLUDE_PATH".to_owned(), value.to_string()))
+    }
+    if let Some(value) = environment.get("CMAKE_LIBRARY_PATH") {
+        desired_env.push(("CMAKE_LIBRARY_PATH".to_owned(), value.to_string()))
+    }
+    let mut paths = vec![];
+    // For each listed dependency, add it to the PATH
+    if let Some(optprefix) = environment.get("HOMEBREW_OPT") {
+        // Comma-separated list of all dependencies in the recursive tree
+        if let Some(dependencies) = environment.get("HOMEBREW_DEPENDENCIES") {
+            for dep in dependencies.split(',') {
+                let short_name = dep.split('/').next().unwrap();
+                let pkg_opt = format!("{optprefix}/{short_name}");
+
+                // Not every package will have a /bin or /sbin directory,
+                // but it's safe to add both to the PATH just in case.
+                paths.push(format!("{pkg_opt}/bin"));
+                paths.push(format!("{pkg_opt}/sbin"));
+            }
+        }
+    }
+    if !paths.is_empty() {
+        let our_path = env!("PATH");
+        let desired_path = format!("{our_path}:{}", paths.join(":"));
+
+        desired_env.insert(0, ("PATH".to_owned(), desired_path));
+    }
+
+    desired_env
+}
+
+// Similar to the above, we read Homebrew's recursive dependency tree and
+// then append link flags to cargo-dist's rustflags.
+// These ensure that Rust can find C libraries that may exist within
+// each package's prefix.
+fn determine_brew_rustflags(base_rustflags: &str, environment: &BTreeMap<&str, &str>) -> String {
+    let mut rustflags = base_rustflags.to_owned();
+    // For each listed dependency, add it to CFLAGS/LDFLAGS
+    if let Some(optprefix) = environment.get("HOMEBREW_OPT") {
+        if let Some(dependencies) = environment.get("HOMEBREW_DEPENDENCIES") {
+            for dep in dependencies.split(',') {
+                let short_name = dep.split('/').next().unwrap();
+                let pkg_opt = format!("{optprefix}/{short_name}");
```

Comment:
it's not a big deal but it does seem slightly odd that we parse this format twice (at very least could be factored to a reusable func)

Reply:
Yeah, I think I can DRY this up a bit.